src/zabapgit_object_devc.prog.abap

```diff
@@ -0,0 +1,556 @@
+*&---------------------------------------------------------------------*
+*&  Include  zabapgit_object_devc
+*&---------------------------------------------------------------------*
+
+CLASS lcl_object_devc DEFINITION
+  INHERITING FROM lcl_objects_super
+  FINAL.
+
+  PUBLIC SECTION.
+    INTERFACES:
+      lif_object.
+    ALIASES:
+      mo_files FOR lif_object~mo_files.
+    METHODS:
+      constructor IMPORTING is_item     TYPE lif_defs=>ty_item
+                            iv_language TYPE spras.
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+    METHODS:
+      get_package IMPORTING iv_devclass       TYPE devclass
+                  RETURNING VALUE(ri_package) TYPE REF TO if_package
+                  RAISING   lcx_exception,
+      update_pinf_usages IMPORTING ii_package    TYPE REF TO if_package
+                                   it_usage_data TYPE scomppdata
+                         RAISING   lcx_exception,
+      set_lock IMPORTING ii_package TYPE REF TO if_package
+                         iv_lock    TYPE abap_bool
+               RAISING   lcx_exception.
+    DATA:
+      mv_local_devclass TYPE devclass.
+ENDCLASS.
+
+CLASS lcl_object_devc IMPLEMENTATION.
+  METHOD constructor.
+    super->constructor( is_item     = is_item
+                        iv_language = iv_language ).
+    mv_local_devclass = is_item-devclass.
+  ENDMETHOD.
+
+  METHOD get_package.
+    cl_package_factory=>load_package(
+      EXPORTING
+        i_package_name             = iv_devclass
+        i_force_reload             = abap_true
+      IMPORTING
+        e_package                  = ri_package
+      EXCEPTIONS
+        object_not_existing        = 1
+        unexpected_error           = 2
+        intern_err                 = 3
+        no_access                  = 4
+        object_locked_and_modified = 5
+        OTHERS                     = 6 ).
+    IF sy-subrc = 1.
+      RETURN.
+    ELSEIF sy-subrc <> 0.
+      lcx_exception=>raise( |Error from CL_PACKAGE_FACTORY=>LOAD_PACKAGE { sy-subrc }| ).
+    ENDIF.
+  ENDMETHOD.
+
+  METHOD lif_object~changed_by.
+    rv_user = get_package( mv_local_devclass )->changed_by.
+  ENDMETHOD.
+
+  METHOD lif_object~compare_to_remote_version.
+    CREATE OBJECT ro_comparison_result TYPE lcl_comparison_null.
+  ENDMETHOD.
+
+  METHOD lif_object~delete.
+    DATA: li_package    TYPE REF TO if_package,
+          lv_changeable TYPE abap_bool.
+
+    RETURN ##TODO.
```

Comment:
just RETURN here, the packages cannot be deleted as part of object deletion, also packages are currently not deleted by abapGit